### PR TITLE
rename docs/source-app & adjust docs links for lightning

### PR DESCRIPTION
**Pull request by Borda**
## What does this PR do?

seems the deploy was failing bease we have not generated `lightning_app`
in fact, we shall be changing all imports in unified docs to `lightning` instead

<details>
  <summary><b>Before submitting</b></summary>

- Was this **discussed/agreed** via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/Lightning-AI/lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- Did you make sure to **update the documentation** with your changes? (if necessary)
- Did you write any **new necessary tests**? (not for typos and docs)
- [x] Did you verify new and **existing tests pass** locally with your changes?
- Did you list all the **breaking changes** introduced by this pull request?
- Did you **update the CHANGELOG**? (not for typos, docs, test updates, or minor internal changes/refactors)

</details>

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing, make sure you have read the [review guidelines](https://github.com/Lightning-AI/lightning/wiki/Review-guidelines). In short, see the following bullet-list:

<details>
  <summary>Reviewer checklist</summary>

- [x] Is this pull request ready for review? (if not, please submit in draft mode)
- [x] Check that all items from **Before submitting** are resolved
- [x] Make sure the title is self-explanatory and the description concisely explains the PR
- [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

</details>

cc @borda @carmocca @akihironitta @justusschock @awaelchli

### CODE REVIEW

File: .github/CODEOWNERS [hunk 1]
```diff
@@ -22,10 +22,10 @@
 /docs/source-pytorch/index.rst              @williamfalcon @lantiga
 /docs/source-pytorch/levels                 @williamfalcon @RobertLaurella
 /docs/source-pytorch/expertise_levels       @williamfalcon @RobertLaurella
-/docs/source-app/                           @williamfalcon @RobertLaurella @tchaton @awaelchli
-/docs/source-app/conf.py                    @borda @awaelchli @carmocca
-/docs/source-app/index.rst                  @williamfalcon @lantiga
-/docs/source-app/expertise_levels           @williamfalcon @RobertLaurella
+/docs/source-lit/                           @williamfalcon @RobertLaurella @tchaton @awaelchli
```
Comment (carmocca) [inline .github/CODEOWNERS:25]: can we call this source-lightning instead?

Comment (akihironitta) [reply .github/CODEOWNERS:25]: or simply `docs/source/`?

**Review by akihironitta — APPROVED** [review]

File: .github/CODEOWNERS (hunk 1 above)
Comment (Borda) [reply .github/CODEOWNERS:25]: source is fine :)

File: docs/source/levels/intermediate/run_lightning_work_in_parallel.rst [hunk 2]
```diff
@@ -3,7 +3,7 @@ Level 6: Run a Lightning component in parallel
 ##############################################
 **Audience:** Users who want to run a Lightning Component in parallel (asynchroneously).
 
-**Prereqs:** You must have finished `Level 5 <debug_a_lightning_app.html>`_.
+**Prereqs:** You must have finished `Level 5 <debug_a_lightning.html>`_.
```
Comment (awaelchli) [inline docs/source/levels/intermediate/run_lightning_work_in_parallel.rst:6]: This is probably not intentional. Can you revert this and also revert renaming the rst file?

File: docs/source/levels/intermediate/index.rst [hunk 3]
```diff
@@ -40,7 +40,7 @@ Learn to coordinate 2+ components into workflows which we call Lightning apps.
 .. displayitem::
    :header: Level 5: Debug a Lightning App
    :description: Learn to debug a lightning app.
-   :button_link: debug_a_lightning_app.html
+   :button_link: debug_a_lightning.html
```
Comment (awaelchli) [inline docs/source/levels/intermediate/index.rst:43]: here too

File: docs/source/levels/intermediate/connect_lightning_components.rst [hunk 4]
```diff
@@ -105,7 +105,7 @@ The next levels does a 2 minute review to make sure you know how to debug a Ligh
 .. displayitem::
    :header: Level 5: Debug a Lightning App
    :description: Learn to debug a lightning app.
-   :button_link: debug_a_lightning_app.html
+   :button_link: debug_a_lightning.html
```
Comment (awaelchli) [inline docs/source/levels/intermediate/connect_lightning_components.rst:108]: here too

File: docs/source/core_api/lightning_app/index.rst [hunk 5]
```diff
@@ -85,7 +85,7 @@ Lightning App API
    :header: LightningApp API
    :description: Look into the Lightning API reference.
    :col_css: col-md-4
-   :button_link: lightning_app.html
+   :button_link: lightning.html
```
Comment (awaelchli) [inline docs/source/core_api/lightning_app/index.rst:88]: same here. and undo renaming the file lightnign_app.rst

File: docs/source/api_reference/components.rst [hunk 6]
```diff
@@ -7,7 +7,7 @@ lightning.app.components
     :local:
     :backlinks: top
 
-.. currentmodule:: lightning_app.components
```
Comment (awaelchli) [inline docs/source/api_reference/components.rst:10]: Careful! this should be `lightning.app` !

File: docs/source/api_reference/api/lightning_app.core.LightningApp.rst [hunk 7]
```diff
@@ -1,6 +1,6 @@
 ﻿.. role:: hidden
     :class: hidden-section
-.. currentmodule:: lightning_app.core
```
Comment (awaelchli) [inline docs/source/api_reference/api/lightning_app.core.LightningApp.rst:3]: same here

File: docs/source/api_reference/api/lightning_app.core.LightningFlow.rst [hunk 8]
```diff
@@ -1,6 +1,6 @@
 ﻿.. role:: hidden
     :class: hidden-section
-.. currentmodule:: lightning_app.core
```
Comment (awaelchli) [inline docs/source/api_reference/api/lightning_app.core.LightningFlow.rst:3]: same here

File: docs/source/api_reference/api/lightning_app.core.LightningWork.rst [hunk 9]
```diff
@@ -1,6 +1,6 @@
 ﻿.. role:: hidden
     :class: hidden-section
-.. currentmodule:: lightning_app.core
```
Comment (awaelchli) [inline docs/source/api_reference/api/lightning_app.core.LightningWork.rst:3]: same here

**Review by awaelchli — CHANGES_REQUESTED** [review]

**Comment by Borda** [comment]
@awaelchli, all good catches, and sorry I missed then... :(

**Comment by Borda** [comment]
the true check is in: https://github.com/Lightning-AI/lightning/actions/runs/4149219807

File: docs/source-app/api_reference/api/lightning_app.core.LightningApp.rst [hunk 10]
```diff
@@ -1,10 +0,0 @@
-﻿.. role:: hidden
```
Comment (awaelchli) [inline docs/source-app/api_reference/api/lightning_app.core.LightningApp.rst:1]: can't find this, where was it moved?

**Review by awaelchli — APPROVED** [review]

File: .pre-commit-config.yaml [hunk 11]
```diff
@@ -71,21 +71,21 @@ repos:
...
-        exclude: docs/source-app
+        exclude: docs/source
 
   - repo: https://github.com/psf/black
     rev: 22.12.0
     hooks:
       - id: black
         name: Format code
-        exclude: docs/source-app
+        exclude: docs/source
```
Comment (carmocca) [inline .pre-commit-config.yaml:81]: Is this intended? Do we not want to just include the app files?

File: docs/source-app/api_reference/api/lightning_app.core.LightningApp.rst (hunk 10 above)
Comment (Borda) [reply docs/source-app/api_reference/api/lightning_app.core.LightningApp.rst:1]: This was removed as it duplicated the generated on and caused an issue while building docs, in fact, that shall be removed a while back as this was a bad artifact...

File: .pre-commit-config.yaml (hunk 11 above)
Comment (Borda) [reply .pre-commit-config.yaml:81]: unfortunalyly it is inted and it was explicitly asked by @williamFalcon as he was not happy about formating docs...
cc: @justusschock

**Pull request merged**